openstack/identity/v3/osinherit/testing/requests_test.go

```diff
@@ -0,0 +1,99 @@
+package testing
+
+import (
+	"testing"
+
+	"github.com/gophercloud/gophercloud/openstack/identity/v3/osinherit"
+	th "github.com/gophercloud/gophercloud/testhelper"
+	"github.com/gophercloud/gophercloud/testhelper/client"
+)
+
+func TestAssign(t *testing.T) {
+	th.SetupHTTP()
+	defer th.TeardownHTTP()
+	HandleAssignSuccessfully(t)
+
+	err := osinherit.Assign(client.ServiceClient(), "{role_id}", osinherit.AssignOpts{
```

Comment:
Shouldn't we also test known failure cases, such as providing both `UserID` and `GroupID`, to ensure the validation works?

Reply:
Done